viewing_party/party.py

```diff
@@ -1,23 +1,161 @@
+from collections import defaultdict
+
 # ------------- WAVE 1 --------------------
 
+
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        ans = {"title": title, "genre": genre, "rating": rating}
+        return ans
+    return None
+
+
+def add_to_watched(user_data, movie):
+    # TODO: Should we check if the movie is already in watched_movies?
+    watched_movies = user_data["watched"]  # watched_movies is a list of dictionaries.
+    watched_movies.append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    # TODO: Should we check if the movie is already in the watchlist?
+    movie_watchlist = user_data["watchlist"] # movie_watchlist is a list of dictionaries.
+    movie_watchlist.append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    movie_watchlist = user_data["watchlist"]
+    # TODO: Should we check if the movie has already been watched?
+    for index, movie in enumerate(movie_watchlist):

```

Comment:
interesting way to loop, using enumerate to get an index!

Reply:
In general, it's considered dangerous practice to modify the content of a data structure that your looping over. This can lead to bugs in more complex code. One alternative could be to find another way without using specific indices, for example, just keeping a copy of a list of movies in movie_watchlist and looping over those.  That way, inside that loop, modifying movie_watchlist would be perfectly safe.